api/v1alpha1/workstatus_types.go

```diff
@@ -31,13 +31,21 @@ type WorkStatus struct {
 
 	Spec   WorkStatusSpec `json:"spec,omitempty"`
 	Status RawStatus      `json:"status,omitempty"`
+	StatusDetails StatusDetails `json:"statusDetails, omitempty"`
 }
 
 // Workstatus spec
 type WorkStatusSpec struct {
 	SourceRef SourceRef `json:"sourceRef,omitempty"`
 }
 
+// StatusDetails spec
+type StatusDetails struct {
+	LastGeneration int64                 `json:"lastGeneration,omitempty"`
+	LastGenerationIsApplied  bool        `json:"lastGenerationIsApplied,omitempty"`
+	LastCurrencyUpdateTime  *metav1.Time `json:"lastCurrencyUpdateTime,omitempty"`
```

Comment:
For something that is optional you would typically mark it so with a stylized comment. See https://github.com/kubestellar/kubestellar/blob/main/api/control/v1alpha1/types.go for examples. Also see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required and examples in the kubernetes project.